### DISCUSSION TITLE
Suicide attacks

### DISCUSSION THREAD

**Discussion by Indy2222** [DigitalExtinction/Game#83] [Ideas]
Make it possible to commit a suicide attack, e.g. crashing a drone into an enemy building.

This would make sense in many circumstances, for example when not having enough battery power to return for a recharge.

**Comment by JackCrumpLeys** [comment]
Maybe this and self destruct on low health. 
We could have a building that can add bombs to units to provide flexible delivery of bombs.
Another idea is units that can latch onto other units as bombs.

Would there be building that can detect these bombs? Can we have devices that specifically target these units from long range? Can we have disarming defenses that don't do damage but remove the bomb?

**Comment by leopeltola** [comment]
Just a side note that suicide units tend to be micro-intensive to use, especially against other units where positioning is essential. Against buildings, it doesn't matter as much.

In addition, "active abilities" where the player presses a button to use an ability are hard to get right. If it's there it should be a viable gameplay option. If eg. all flying units can be crashed into buildings is the best play to manually crash every unit once it's near dying? This could be fun when there aren't that many units but can incentivize activities that aren't fun in the late game. Managing self-destructs for hundreds of units at a time to get an advantage isn't necessarily fun, but players can feel pressured to do it regardless.

**Reply by Indy2222** [reply]
Thank you for the feedback! You are right.

I have some ideas about automation bud I have not yet put it down to text / sketches anywhere. The main idea is to allow the user do define simple rules / a tree of condition based actions to automate stuff. This should shift the focus from micro management to strategy. 

Examples:
* automatic attack once enough units amass,
* automatic manufacturing when number of specific units go below a number,
* suicide attacks under certain conditions.

For obvious reasons, this will be extremely hard to pull off. It must be simple to use and the UX well designed. At the same time, it must be capable enough so it is useful as it must come close to micro-management options.

**Reply by leopeltola** [reply]
Sounds good, though difficult! 

Sensible defaults/presets for the automation stuff would go a long way. Something that could also be convenient is being able to import and export the rules as text to make them easier to share with others on forums etc. 

Thanks for your work on the game! I've been playing around with rts prototypes for some time now but it's great to see others work on the genre too, especially open source! Inspires me to work on my own project too.

**Reply by Indy2222** [reply]
> Sensible defaults/presets for the automation stuff would go a long way. Something that could also be convenient is being able to import and export the rules as text to make them easier to share with others on forums etc.

Agreed.

> Thanks for your work on the game! I've been playing around with rts prototypes for some time now but it's great to see others work on the genre too, especially open source! Inspires me to work on my own project too.

:heart:

We would welcome any contributions from you! Feedback, art, bug reports, pull requests...

**Reply by leopeltola** [reply]
> We would welcome any contributions from you! Feedback, art, bug reports, pull requests...

I'll try! I'm still getting comfortable with Rust so code is hard but I'll do my best to drop by every now and then. Or maybe I'll try my hand at one of the `Good-First-Issue`'s, it could be a good way to learn :)

Is there a list of needed art assets anywhere? Having a dedicated page listing all the UI assets, 3D models, etc. would make it easier for artists to know what to make. Perhaps tagged with priority or equivalent. Additionally, it could include specifications such as file formats, recommended polygon counts for units, texture sizes, and so on.

**Reply by Indy2222** [reply]
> > We would welcome any contributions from you! Feedback, art, bug reports, pull requests...
> 
> I'll try! I'm still getting comfortable with Rust so code is hard but I'll do my best to drop by every now and then. Or maybe I'll try my hand at one of the `Good-First-Issue`'s, it could be a good way to learn :)

Yes, please try to find one of the Good-First-Issues.

> Is there a list of needed art assets anywhere? Having a dedicated page listing all the UI assets, 3D models, etc. would make it easier for artists to know what to make. Perhaps tagged with priority or equivalent. Additionally, it could include specifications such as file formats, recommended polygon counts for units, texture sizes, and so on.

Every active object has an associated JSON with a collider and other coordinates. The collider should have a few triangles (small objects) to several dozens or more (buildings). Please see 
https://docs.de-game.org/objects/index.html

Please also see this assets contributing guideline (which we will adapt as we learn) https://github.com/DigitalExtinction/Game/blob/main/CONTRIBUTING.md#contributing-assets

Here is the issue tracking the initial design of 3D assets https://github.com/DigitalExtinction/Game/issues/238 The most urgent ones are attacker drone & base building.
